### ISSUE TITLE
Hotkey/UI button to change priority selection level on any unit.

### ISSUE THREAD

**Issue by WhoCaresora** [OpenRA/OpenRA#19195]
It has always been a cause of frustration to see some units you dedicated to a specific location/task, tag along with your army because at one point they were caught in a square selection box.

In order to remedy to those frustrations and to give the game another layer of micro/possibilities, I would like to suggest the implementation of a hotkey/UI button to be able to toggle a unit's priority selection between High and Low. I believe there is more than 1 level of selection priority but this feature requires being high or not high, nothing more

Here some examples on how it could be used or be beneficial : 

- spotter riflemen : switch to low priority to avoid being selected with army and stays where he was sent so be a passive scout.
- medic & mechanic : switch to low priority to use them with guard order and not losing the order by being selected with the army 
- ore field guard detachment : switch to low priority to be sure they are not selected with nearby unit and stays on duty
- aircraft : switch to low priority when used alongside an army to facilitate selection of both ground and air unit independently.
-ore truck : switch to high priority when used as tank in front of infantry to be selected easier alongside the army

There should be a lot more of beneficial gameplay possibilities that i don't think about or that would be discovered when and if the feature is live.

I don't expect any side effect as all units should keep their default selection priority on being built. New or casual player would not be bothered by that as it requires an action to take effect. But once they find out the mechanism they can use it too.

In addition to this feature, it can be made that a unit sent to be repaired automatically goes in low priority and is back on high priority when exiting the SD  (#17537). The player could at any moment reverse the priority if needed.

I hope a discussion here would bring more polish to my request and hopefully a way to be integrated to the game at one point.

**Comment by ParselyBunny** [comment]
Use of control groups can solve the problems of mixing in non-combat units in a lot of situations, but I could see myself using this when adding mechanics, ore trucks, or medics to an army.

**Comment by WhoCaresora** [comment]
non combat units are not part of the meta in competitive play because the level of micro needed to keep them safe (by using group control as an option) is not worth it. But the day you can put them in low priority selection, make some queued guard order on the army, it will change. After the short micro as I described, you are free to select and move your army at will, and the support unit will just follow the army because of the guard order without putting themself in danger. No more micro needed past the "setting point" and you can benefit fully of the healing, jamming, repairing ect.

**Comment by CortexReaver** [comment]
I agree that selecting correct type of units during fight is a problem, but not sure about how this is gonna work "in the field". There was also someone suggesting UI icons corresponding with each unit type (air, ground, infantry) for selection filtering. And I am all for using both features (priority filtering AND unit type filtering), but how it will look and more importantly work? Will they not break each other? Sure it will cause a ton of bugs.
What's about a hypothetical situation where you have more low priority units and less high priority, but someone is attacking and you need to defend yourself quick by selecting all available units? For medics and mechanics its easier: making them "low priority" if they are linked to someone.
Also is a good idea to keep interface clean during battle. Perhaps, optional interface icons for "pros"? Unit type filtering OR priority filtering, priority filtering AND unit type filtering, none of them - four options.
